dotnet/src/InternalUtilities/src/System/EnumerableExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Provides extension methods for <see cref="IEnumerable{T}"/> interface.
+/// </summary>
+[ExcludeFromCodeCoverage]
+internal static class EnumerableExtensions
+{
+    public static bool IsNotEmpty<T>(this IEnumerable<T> enumerable) =>
+        enumerable is not ICollection<T> collection || collection.Count != 0;
```

Comment:
This code may not work with custom collections - `public class CustomCollection<T> : IEnumerable<T>`

Reply:
Agreed, I removed this file and moved it to `Kernel.cs` as private helper method.